src/lib/xmlparse.rs

```diff
@@ -7861,14 +7985,16 @@ impl XML_ParserStruct {
         &mut self,
         mut enc_type: EncodingType,
         mut buf: ExpatBufRef,
-    ) -> *mut AttributeId {
+    ) -> Ptr<AttributeId> {
         let mut dtd_tables = self.m_dtd.tables.borrow_mut();
+        // Convert the RefMut manually into a &mut
+        // so the compiler can split borrows over the fields
```

Comment:
Does this comment belong here? `dtd_tables` doesn't get turned into a `&mut`? (Which I assume would be `&mut *dtd_tables`?)